src/nbl/video/CVulkanDescriptorPool.cpp

```diff
@@ -25,4 +25,24 @@ void CVulkanDescriptorPool::setObjectDebugName(const char* label) const
 	nameInfo.pObjectName = getObjectDebugName();
 	vkSetDebugUtilsObjectNameEXT(vulkanDevice->getInternalObject(), &nameInfo);
 }
+
+bool CVulkanDescriptorPool::freeDescriptorSets_impl(const uint32_t descriptorSetCount, IGPUDescriptorSet* const* const descriptorSets)
+{
+    constexpr auto MaxDescriptorSetCount = 4u;
+    assert(descriptorSetCount <= MaxDescriptorSetCount);
+    VkDescriptorSet vk_descriptorSets[MaxDescriptorSetCount];
```

Comment:
nope, just use a vector, the 4 constant means nothing here

There's only 4 slots to bind to PIPELINE BIND POINTS in the engine, but DS Pool can have as many as it wants sitting inside cause its only a storage provider

Reply:
still a TODO

Reply:
> There's only 4 slots to bind to PIPELINE BIND POINTS in the engine, but DS Pool can have as many as it wants sitting inside cause its only a storage provider

Right, it was stupid of me.

Reply:
Resolved.